model/task/task.go

```diff
@@ -3669,6 +3669,58 @@ func GetGroupedTaskStatsByVersion(versionID string, opts GetTasksByVersionOption
 
 }
 
+// GetMatchingBaseTasks returns the corresponding base tasks for tasks on a version.
+// If a task ran on the base version but was not scheduled on the version, then it won't be returned.
+func GetMatchingBaseTasks(versionID string, baseVersionID string) ([]Task, error) {
+	pipeline := []bson.M{}
+
+	pipeline = append(pipeline,
+		bson.M{"$match": bson.M{VersionKey: baseVersionID}},
+	)
+
+	// Look up and filter out tasks that don't exist on version.
+	VersionTasksKey := "VersionTasks"
+	taskMatch := []bson.M{
+		{"$eq": []string{"$" + BuildVariantKey, "$$" + BuildVariantKey}},
+		{"$eq": []string{"$" + DisplayNameKey, "$$" + DisplayNameKey}},
+	}
+	pipeline = append(pipeline, []bson.M{
+		{"$lookup": bson.M{
```

Comment:
`$lookups` are inherently slower and we will notice a perf hit on larger versions. Since we already know the both of the versions we can can accomplish the same thing much more performantly by using a `$group` instead.
1. `$match` on both version ids
2. `$match` on `ActivatedTimeKey: {"$ne": utility.ZeroTime}`
3. `$group` tasks together by buildvariant and displayname
4. `$unwind` the tasks and get their status
5. `$profit`

Reply:
Hmm so I'm not sure if this can be straightforward-ly accomplished without a `$lookup`, but I did push up a version of the pipeline which is what I'd imagine it would look like without a lookup (heavily annotated to explain what's going on)